pkg/adafruitio/adafruitio.go

```diff
@@ -26,17 +27,19 @@ func New(username, AIOKey string) (*Client, error) {
 	}
 
 	// Check client credentials.
-	req, err := http.NewRequest(http.MethodGet, "https://io.adafruit.com/api/v2/user", nil)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not construct API request")
-	}
-	res, err := client.send(req)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not send API request")
-	}
-	err = check(res)
-	if err != nil {
-		return nil, errors.Wrap(err, "API response has error")
+	if client.aioKey != "" {
```

Comment:
I don't like this implementation change. I think it makes the function's contract unclear.

What does it mean to provide an empty key to an API client constructor? Normally, I would think this is an error, but this implementation change makes it a valid behaviour. This creates a footgun since someone that messes up a function call (e.g. intend to create an authenticated client, but forget to correctly pass a key) could reasonably expect that accident to return an error rather than continuing silently.

It also causes other kinds of nonsensical behaviour e.g. requests are now sent with an empty `X-AIO-Key` header.

I think it would be much clearer keep the credential checking behaviour and instead provide another function for creating unauthenticated clients.